### PR TITLE
grp is no longer imported unconditionally

### LINKED ISSUE #6797: ModuleNotFoundError: No module named 'grp'
# Checklist

- [x] I have verified that the issue exists against the `master` branch of Celery.
- [ ] This has already been asked to the [discussion group](https://groups.google.com/forum/#!forum/celery-users) first.
- [x] I have read the relevant section in the
  [contribution guide](http://docs.celeryproject.org/en/latest/contributing.html#other-bugs)
  on reporting bugs.
- [x] I have checked the [issues list](https://github.com/celery/celery/issues?q=is%3Aissue+label%3A%22Issue+Type%3A+Bug+Report%22+-label%3A%22Category%3A+Documentation%22)
  for similar or identical bug reports.
- [x] I have checked the [pull requests list](https://github.com/celery/celery/pulls?q=is%3Apr+label%3A%22PR+Type%3A+Bugfix%22+-label%3A%22Category%3A+Documentation%22)
  for existing proposed fixes.
- [x] I have checked the [commit log](https://github.com/celery/celery/commits/master)
  to find out if the bug was already fixed in the master branch.
- [x] I have included all related issues and possible duplicate issues
  in this issue (If there are none, check this box anyway).

Hey there! I've read all the checkboxes etc. but it's such an obvious bug, that I feel they are not necessary. I hope that this unconformity won't let you just ignore the actual issue:

# Steps to Reproduce

## Minimally Reproducible Test Case
Run celery on Windows, no matter which python.

### Expected Behavior
It should start

### Actual Behavior
```
  File ".venv\lib\site-packages\celery\platforms.py", line 9, in <module>
    import grp
ModuleNotFoundError: No module named 'grp'
```
The module `grp` is only available on linux, thus this does not work on widnwos and also not make sense as a global import. It's also imported below by `try_import`.

### Possible fix
This line should simply be removed, because below there is a `try_import` for this as well. Also in the `check_privileges`, the non-existence of `grp` should be checked. Unfortunately due to my companies restrictions I cannot submit a MR.

Git blame shows @namloc2001 and @thedrow for [this line which is still on master](https://github.com/celery/celery/blob/ce567e31065e3361493ebb33a23e2f04c07cc371/celery/platforms.py#L9).

Thanks!

### PR DESCRIPTION
**Pull request by thedrow**
*Note*: Before submitting this pull request, please review our [contributing
guidelines](https://docs.celeryproject.org/en/master/contributing.html).

## Description

This fixes a regression introduced in #6600 which caused an import error on non-Unix platforms.

Fixes #6797.

### CODE REVIEW

**Comment by thedrow** [comment]
@voegtlel Could you please verify this patch works for you?

**Comment by thedrow** [comment]
The test doesn't seem to cover our new code branch.
Any suggestions on how to make the modules disappear?

File: t/unit/utils/test_platforms.py [hunk 1]
```diff
@@ -1001,3 +1009,11 @@ def test_check_privileges_with_c_force_root_and_no_group_entry(accept_content, r
             check_privileges(accept_content)
 
         assert recwarn[0].message.args[0] == ASSUMING_ROOT
+
+
+def test_skip_checking_privileges_when_grp_is_unavailable(recwarn):
+    with patch('celery.platforms.try_import',
+               return_value=None):
```
Comment (maybe-sybr) [inline t/unit/utils/test_platforms.py:1016]: Probably `patch("celery.platforms.grp", new=None)` and just rely on magic knowledge of the `grp` name - I can't think of a practical alternative that wouldn't cause more potential confusion due to reloading the module and the heavy use of from imports in the celery codebase.

**Review by maybe-sybr — CHANGES_REQUESTED** [review]

**Comment by maybe-sybr** [comment]
The change itself lgtm

**Review by auvipy — APPROVED** [review]

**Review by maybe-sybr — APPROVED** [review]
lgtm

**Comment by auvipy** [comment]
should be on 5.1.1 as it fixes a regression O:)

**Pull request merged**